Principal Software Engineer (#30)
OctopusDeploy/People: Software-Engineering.md

- I can confidently and charismatically pitch ideas, positively influencing my team to take decisive action based on reaching consensus.
- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.

## Principal Software Engineer

> > I am an influential Engineering [leader](Leadership.md) who identifies problems and opportunities relevant to Octopus' strategy, builds consensus to invest in solutions, and guides Engineering to deliver significant lasting change across Octopus.

Reviewer: Should we mention "Mission"s here, and give "mission"s a clearer definition? I think it would aid in clarity.

Reviewer 2: Concensus was that adding terminology around "missions" here doesn't improve the position description, so we will opt not to.

> ### I have a knack for uncovering important problems and opportunities that are critical to the long-term success of Octopus

Reviewer: ```suggestion
### I uncover important problems and opportunities that are critical to the long-term success of Octopus
```

> - I have a deep understanding of Octopus products and our strategy

Reviewer: If a principal engineer is working in the Octopus Server core team, how deep does their knowledge need to be in areas covered by other teams?

Reviewer: Is there something to be said along the lines of "a principal engineer needs to be across all of the major things happening at Octopus?"

Reviewer 2: ```suggestion
- I have a deep understanding of Octopus products and strategy
- I collaborate with other Engineering leaders to ensure I am aware of all major initiatives at Octopus and can account for them in my own
```

Reviewer 3: (Small typo btw? `in my own`)

Reviewer 2: Nope, perhaps it needs the word "initiatives" on the end again...

Reviewer 3: Ah, right 🙈 Yup, I think `initiatives` at the end helps, thanks 🙇

> - I identify significant problems, inhibitors and opportunities, and advocate for solutions that align our strategy and Engineering efforts

Reviewer 2: ```suggestion
- I have a track record of identifying significant problems, inhibitors and opportunities, and for creating solutions that align our strategy and Engineering efforts, and that prove over time to be the right solutions
```

Reviewer 2: Resolves https://github.com/OctopusDeploy/People/pull/30/files/44987a39337bb1d6817693c762c4fc34531a374f#r533875494

> - I collaborate with others to articulate solutions, positively influence and convince that we should fund their implementation

Reviewer: I think there are two things here - maybe we can say them both explicitly?
- Principal engineers have a good track record of identifying the important problems and solutions that tend to be funded
- Principal engineers are good at persuading the organisation/betting table/etc. to fund these pitches/missions/etc.

Reviewer 2: I've improved part of the above, but I think there is an item missing regarding working with product that your second point helps steer us toward. I'll see where is best to add that item.

> - My contributions go beyond patching or extending existing solutions, which may not be fit for purpose anymore. They open up whole new areas of strategic possibility, to take our Engineering team, and Octopus as a whole, to the next level.

Reviewer: Is there a better way to say `to the next level`?

Reviewer 4: @andrewabest will wordsmith. :)

Paul loves "world-class".

Reviewer 2: ```suggestion
- My contributions go beyond patching or extending existing solutions, which may not be fit for purpose anymore. They open up whole new areas of strategic possibility that progress our journey of making Octopus a world-class Engineering organization.
```

> ### I am trusted to take care of a large, critical or complex strategic area or product

Reviewer: What does "take care of" mean?

Reviewer: I was reflecting on the new team shape for the actions team: PSE + EM + PM forming the leadership group for that team. It feels like this team would be collectively responsible for the outcomes of that team. I think that is what "take care of" means in this context. Is "accountable" a better word?

```suggestion
### I am accountable for a large, critical or complex strategic area or product
```

Reviewer 2: I like this @TomPeters - if we combine it with one of @michaelnoonan's 🌱 's below, and add a bit of Hemmingway, we get:

```suggestion
### I am the custodian of an area of strategic importance to Octopus
```

> - I am comfortable taking on a major challenge that requires building or re-architecting an important part of the Octopus ecosystem (e.g. Modularity, or Octopus Cloud)

Reviewer 3: Suggest phrasing such that new starters can understand, e.g.:

```suggestion
- I am comfortable taking on a major challenge that requires building or re-architecting an important part of the Octopus ecosystem (e.g. re-architecting Octopus Server to allow for adding new step types as modular extensions, or creating a completely new Octopus Cloud platform)
```

Reviewer: What does "important" mean?

Reviewer 2: I think we can eliminate some of the verbosity here, how about:
 
```suggestion
- I am comfortable taking on large-scale greenfields and brownfields initiatives (e.g. re-architecting Octopus Server to allow for adding new step types as modular extensions, or creating a completely new Octopus Cloud platform)
```

It still doesn't clarify what 'large-scale' means, but I think its ok if that is qualitative.

> - I identify people and teams with the necessary skills, work out how to fill gaps, and guide and influence these people and teams collaboratively

Reviewer 2: ```suggestion
- I identify people and teams with the necessary skills, work out how to fill gaps, and guide and influence these people and teams collaboratively
- I use a variety of tools and techniques to effectively communicate short and long-term plans and goals to my teams, stakeholders, Octopus senior leadership, and the broader organization
```

> - I use my Engineering experience to collaboratively steer teams toward good long-term decisions along the way

Reviewer: ```suggestion
- I use my Engineering experience to collaboratively steer teams toward good long-term decisions
```

- I can confidently and charismatically pitch ideas, positively influencing my team to take decisive action based on reaching consensus.
- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.

## Principal Software Engineer

> I am an influential Engineering [leader](Leadership.md) who identifies problems and opportunities relevant to Octopus' strategy, builds consensus to invest in solutions, and guides Engineering to deliver significant lasting change across Octopus.

### I uncover important problems and opportunities that are critical to the long-term success of Octopus

- I have a deep understanding of Octopus products and strategy
- I collaborate with other Engineering leaders to ensure I am aware of all major initiatives at Octopus and can account for them in my own initiatives
- I have a track record of identifying significant problems, inhibitors and opportunities, and for creating lasting, best-fit solutions that align our strategy and Engineering efforts

> - I collaborate with others to articulate solutions, positively influence and convince that we should fund their implementation

Reviewer 2: ```suggestion
- I work closely with product, engineering, and executive stakeholders to articulate and refine solutions to problems that are timely, important, and impactful. 
```

Reviewer 2: I think this could be further improved, but I'll use this as a start.

- My contributions go beyond patching or extending existing solutions, which may not be fit for purpose anymore. They open up whole new areas of strategic possibility that progress our journey of making Octopus a world-class Engineering organization.

### I am the custodian of an area of strategic importance to Octopus

- I am comfortable taking on large-scale greenfields and brownfields initiatives (e.g. re-architecting Octopus Server to allow for adding new step types as modular extensions, or creating a completely new Octopus Cloud platform)
- I identify people and teams with the necessary skills, work out how to fill gaps, and guide and influence these people and teams collaboratively
- I use a variety of tools and techniques to effectively communicate short and long-term plans and goals to my teams, stakeholders, Octopus senior leadership, and the broader organization
- I use my experience to collaboratively steer teams toward good long-term technical decisions
- The results of the initiatives I champion are measurable

> 

Reviewer 2: ```suggestion

### I share my knowledge both within Octopus, and with the Octopus and wider developer community

- I actively blog about the lessons we are learning as we improve engineering excellence to clarify our thinking, reinforce the learning, attract customers and potential future employees.
```

- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.

## Principal Software Engineer

> I am an influential Engineering [leader](Leadership.md) who identifies problems and opportunities relevant to Octopus' strategy, builds consensus to invest in solutions, and guides Engineering to deliver significant lasting change across Octopus.

### I uncover important problems and opportunities that are critical to the long-term success of Octopus

- I have a deep understanding of Octopus products and strategy
- I collaborate with other Engineering leaders to ensure I am aware of all major initiatives at Octopus and can account for them in my own initiatives
- I have a track record of identifying significant problems, inhibitors and opportunities, and for creating lasting, best-fit solutions that align our strategy and Engineering efforts
- I work closely with product, engineering, and executive stakeholders to articulate and refine solutions to problems that are timely, important, and impactful. 

> - My contributions go beyond patching or extending existing solutions, which may not be fit for purpose anymore. They open up whole new areas of strategic possibility that progress our journey of making Octopus a world-class Engineering organization.

Reviewer 4: "They" in the second sentence caused me to pause for a moment due to ambiguity. Is this "they" the person... oh, "they" is the contribution. I've made a small suggestion to clarify:
```suggestion
- Instead of simply patching or extending existing solutions, which may not be fit for purpose anymore, my contributions open up whole new areas of strategic possibility that progress our journey of making Octopus a world-class Engineering organization.
```

### I am the custodian of an area of strategic importance to Octopus

- I am comfortable taking on large-scale greenfields and brownfields initiatives (e.g. re-architecting Octopus Server to allow for adding new step types as modular extensions, or creating a completely new Octopus Cloud platform)
- I identify people and teams with the necessary skills, work out how to fill gaps, and guide and influence these people and teams collaboratively
- I use a variety of tools and techniques to effectively communicate short and long-term plans and goals to my teams, stakeholders, Octopus senior leadership, and the broader organization
- I use my experience to collaboratively steer teams toward good long-term technical decisions
- The results of the initiatives I champion are measurable

### I share my knowledge both within Octopus, and with the Octopus and wider developer community

> - I actively blog about the lessons we are learning as we improve engineering excellence to clarify our thinking, reinforce the learning, attract customers and potential future employees

Reviewer 4: I want to add two new examples/points, one about an internally-focused impact as a partner to external (blogging), and one about expertly tailoring communication to the target audience. To me, these seem like key differentiators of PSE from LSE.

```suggestion
- I actively influence engineering teams across Octopus resulting in improved engineering excellence and overall productivity
- I actively blog about the lessons we are learning as we improve engineering excellence to clarify our thinking, reinforce the learning, attract customers and potential future employees
- I have a proven track record of expertly tailoring my communication to the target audience with a result of increased clarity and impact

```
